grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -51,14 +60,546 @@ class ApiGetOpenApiDescriptionHandler(api_call_handler_base.ApiCallHandler):
   args_type = None
   result_type = ApiGetOpenApiDescriptionResult
 
+  def __init__(self, router):
+    self.router = router
+    self.open_api_obj = None # The main OpenAPI description object.
+    self.schema_objs = None
+
+    self.proto_primitive_types_names = {
+      protobuf2.TYPE_DOUBLE: "protobuf2.TYPE_DOUBLE",
+      protobuf2.TYPE_FLOAT: "protobuf2.TYPE_FLOAT",
+      protobuf2.TYPE_INT64: "protobuf2.TYPE_INT64",
+      protobuf2.TYPE_UINT64: "protobuf2.TYPE_UINT64",
+      protobuf2.TYPE_INT32: "protobuf2.TYPE_INT32",
+      protobuf2.TYPE_FIXED64: "protobuf2.TYPE_FIXED64",
+      protobuf2.TYPE_FIXED32: "protobuf2.TYPE_FIXED32",
+      protobuf2.TYPE_BOOL: "protobuf2.TYPE_BOOL",
+      protobuf2.TYPE_STRING: "protobuf2.TYPE_STRING",
+      protobuf2.TYPE_BYTES: "protobuf2.TYPE_BYTES",
+      protobuf2.TYPE_UINT32: "protobuf2.TYPE_UINT32",
+      protobuf2.TYPE_SFIXED32: "protobuf2.TYPE_SFIXED32",
+      protobuf2.TYPE_SFIXED64: "protobuf2.TYPE_SFIXED64",
+      protobuf2.TYPE_SINT32: "protobuf2.TYPE_SINT32",
+      protobuf2.TYPE_SINT64: "protobuf2.TYPE_SINT64",
+    }
+    self.primitive_types_names = \
+      list(self.proto_primitive_types_names.values()) \
+      + ["BinaryStream",]
+
+  def _SimplifyPathNode(self, node: str) -> str:
```

Comment:
SimplifyPathComponent

(In /a/b/c the "parts" are called components).

Please change all occurrences of "node" to "component"

Also, it doesn't really "simplify", but it normalizes/translates?

Reply:
I made the change from `Node` -> `Component` and from `Simplify` -> `Normalize`. Thank you for suggesting this! The reason I used "simplify" was because the Werkzeug URL may contain type converters which are eliminated in the resulting nodes and path, but I believe that the term "normalize" should cover this semantic as well as the idea that we're replacing "<>"s with "{}"s.